plugins/woocommerce/src/Internal/DownloadPermissionsAdjuster.php

```diff
@@ -99,7 +99,21 @@ public function adjust_download_permissions( int $product_id ) {
 
 		$children_with_downloads = array();
 		foreach ( $children_ids as $child_id ) {
-			$child                                = wc_get_product( $child_id );
+			$child = wc_get_product( $child_id );
+
+			// Ensure we have a valid child product.
+			if ( ! $child ) {
```

Comment:
Would it be possible here to check for the expected return value instead of the failure value? That would be more precise given that the bug this addresses relates to the return value not being an object with a specific method.

Maybe something like

```suggestion
			if ( ! $child instanceof WC_Product ) {
```

Reply:
Yes, I think that makes sense (especially since the next method to be called is `get_download_files_and_permissions()`, which expects a `WC_Product`). Will import the class and apply this change :+1:

Reply:
Updated in https://github.com/woocommerce/woocommerce/pull/36431/commits/9bfd8535f7c15cb1eb528e028ed5a9571512a9bc.